sensor/api/api_views.py

```diff
@@ -127,21 +135,46 @@ def put(self , request , *args, **kwargs):
 class LocationListView(generics.ListCreateAPIView):
     queryset = models.Location.objects.all()
     serializer_class = LocationSerializer
-    
 
-class LocationView(generics.RetrieveAPIView):    
+
+class LocationView(generics.RetrieveAPIView):
     queryset = models.Location.objects.all()
     serializer_class = LocationSerializer
 
+class LocationCreator(View):
+
+    def get(self, request, pk):
+        device = get_object_or_404(Device, pk=pk)
+        device_data = DeviceSerializer(data=device)
+        if 'key' not in request.GET:
+            raise KeyError('Missing API-key "key".  Needed to authenticate.')
+        authenticate(device, request.GET['key'])
+
+        for k in ('latitude', 'longitude', 'name'):
+            if k not in request.GET:
+                raise KeyError('Missing key %s' % k)
+        payload = {'key': request.GET['key'],
+                   'latitude': request.GET['latitude'],
+                   'longitude': request.GET['longitude'],
+                   'name': request.GET['name'],
+                   'altitude': 0,  # optional, set if exists
+        }
+        if 'altitude' in request.GET:
+            payload['altitude'] = request.GET['altitude']
+        loc = Location.create(payload)
+        device.location = loc
+        device.save()
+        return HttpResponseRedirect('/sensor/view/device/%s/' % device.id)
```

Comment:
Try to use the `reverse` functionality to look up a named url here, that way the url structure is *only* configured in the `urls.py` scheme - and not scattered around in the code.